lib/backend/k8s/resources/customresource_test.go

```diff
@@ -107,11 +108,25 @@ var _ = Describe("BGP Peer conversion methods", func() {
 		Expect(r.GetObjectMeta().GetResourceVersion()).To(Equal(res1.ObjectMeta.ResourceVersion))
 		Expect(r).To(BeAssignableToTypeOf(&apiv2.BGPPeer{}))
 		Expect(r.(*apiv2.BGPPeer).Spec).To(Equal(res1.Spec))
+
+		// Make sure to clean up the annotations on the resource
+		err = ConvertK8sResourceToCalicoResource(r)
+		Expect(err).NotTo(HaveOccurred())
 	})
 
 	It("should convert between a Kuberenetes resource and the equivalent KVPair", func() {
+		// Make sure it returns the same resource if a resource is given without Calico metadata annotations.
 		kvp, err := client.convertResourceToKVPair(res1)
 		Expect(err).NotTo(HaveOccurred())
+		Expect(kvp.Value.(*apiv2.BGPPeer)).To(BeAssignableToTypeOf(&apiv2.BGPPeer{}))
```

Comment:
I think you mean this to be `Expect(kvp.Value).To(BeAssignableToTypeOf(&apiv2.BGPPeer{}))`

i.e. without casting the Value - otherwise if it isn't a BGPPeer that cast will panic.

Reply:
Given you are doing a value comparison below - I don't think this particular Expect adds any value.